app/controllers/api/projects.php

```diff
@@ -188,33 +217,19 @@
                 continue;
             }
 
-            $attributes = [];
-            $indexes = [];
-
-            foreach ($collection['attributes'] as $attribute) {
-                $attributes[] = new Document([
-                    '$id' => $attribute['$id'],
-                    'type' => $attribute['type'],
-                    'size' => $attribute['size'],
-                    'required' => $attribute['required'],
-                    'signed' => $attribute['signed'],
-                    'array' => $attribute['array'],
-                    'filters' => $attribute['filters'],
-                    'default' => $attribute['default'] ?? null,
-                    'format' => $attribute['format'] ?? ''
-                ]);
-            }
+            $attributes = \array_map(function (array $attribute) {
+                return new Document($attribute);
+            }, $collection['attributes']);
+
+            $indexes = \array_map(function (array $index) {
+                return new Document($index);
```

Comment:
are `default` and `format` handled appropriately when using the `Document()` constructor directly ?

Reply:
Seems like it, the default check is redundant, it's coalescing to null if the value is null, and can't see anywhere it would matter if `format` is null instead of empty. All tests are passing too